Cabal-syntax/src/Distribution/Types/InstalledPackageInfo.hs

```diff
@@ -92,7 +92,10 @@ data InstalledPackageInfo
         frameworks        :: [String],
         haddockInterfaces :: [FilePath],
         haddockHTMLs      :: [FilePath],
-        pkgRoot           :: Maybe FilePath
+        pkgRoot           :: Maybe FilePath,
+        -- Artifacts included in this package:
+        providesStaticArtifacts  :: Bool,
```

Comment:
These need more documentation. What do we mean by "artifacts" here? How do these relate to GHC's notion of ways? Why are "static" and "dynamic" special (e.g. what about profiling)?

Reply:
Ben: this is discussed in the prior two PRs -- https://github.com/haskell/cabal/pull/8624 and https://github.com/haskell/cabal/pull/8461

Initially it had added five fields, not just two. However, the other part of the split PR only now makes use of these two, so the scope was scaled back for simplicity. (Unfortunately, this isn't in the history of the linked PR due to it being force-pushed). I think extending the logic to profiling (and teaching the solve about that) would also be useful, but its out of scope for the current round of work, so having those fields added without then being used opened the way to some bikeshedding and confusion which this hopefully avoids.

Reply:
@ben: do you mean documentation in the code included in GHC so that GHC devs don't need to rummage in cabal PRs? That makes sense. Or at least ready links to the aformentioned PRs from the code comments. Or both.

Reply:
Based on common practice in GHC, I suspect the answer is "document it right there in the code." I think it's a best practice in any case. :) 

(Wrong `@ben` fyi)

Reply:
And I'm sorry @ben. :)

Reply:
@chreekat hit the nail on the head. Yes, the documentation should be next to the implementation otherwise future us has little chance of finding it.

Reply:
Fair enough. I should have been more specific when asking @bairyn for the artificial commit, but I lacked foresight. Let wait with resuming the 3.10 semi-freeze and publishing a new dogfooding tag till today's devs meeting. I know @bairyn is busy, so that probably means the PRs have to wait until GHC 9.8. Tough luck.